### PR TITLE
Temporary lower command-prompt when executing the command

### LINKED ISSUE #1194: spotclean doesn't work from the in-game terminal
Selecting a tile with `k` and attempting to run `spotclean` with the in-game terminal, [command-prompt](https://dfhack.readthedocs.io/en/stable/docs/Plugins.html#id232), fails with the following error:

```
Could not invoke spotclean: unsuitable UI state.
```

Typing `spotclean` into the actual command prompt succeeds, naturally. It seems like the presence of the in-game command prompt is incorrectly recognized as an unsuitable UI state.

### PR DESCRIPTION
**Pull request by suokko**
command-prompt viewscreen may affect command execution if they are
looking for UI state. To make commands execute simillary to hotkeys or
console commands the viewscreen needs to removed from the top position.

Fixes #1194

I'm unsure if LowerScreen should be part of Core or module/Screen. I first put it to the Core but I can moved it to Screen if it sounds like a better place for this functionality.

What about lowering position? I only lower it a level but it might be better to lower to bottom (gview->view.child = screen_)

### CODE REVIEW

File: library/include/Core.h [hunk 1]
```diff
@@ -198,6 +198,14 @@ namespace DFHack
 
         static void cheap_tokenise(std::string const& input, std::vector<std::string> &output);
 
+        //! Temporary lower a screen until constructor is called
```
Comment (lethosor) [inline library/include/Core.h]: constructor -> destructor?

File: library/Core.cpp [hunk 2]
```diff
@@ -2769,6 +2769,37 @@ std::string Core::GetAliasCommand(const std::string &name, const std::string &de
...
+    df::viewscreen* bc = b->child;
+
+    ap->child = b;
+    b->parent = ap;
+
+    a->parent = b;
+    b->child = a;
+
+    if (bc) bc->parent = a;
+    else Core::getInstance().top_viewscreen = a;
```
Comment (lethosor) [inline library/Core.cpp]: Does anything important actually use this? I was under the impression that things like spotclean used getCurViewscreen(), which doesn't use this.

Comment (suokko) [reply library/Core.cpp]: Gui::cursor:hotkey() gets it from Plugin::invoke() which uses Core::getTopViewscreen()

**Comment by suokko** [comment]
Completely unlinking sounds safer. I will change to that. If I set links carefully I think that any stale pointers to an old top viewscreen would still handle it sanely. I just need to figure if keeping screen->parent  or setting it to null is safer.

**Comment by lethosor** [comment]
Also, forgot to mention it, but I think this belongs in the screen module. An exception might be if you definitely need to work with core internals, like top_viewscreen, but even then I might prefer just allowing the thing in the screen module to deal with top_viewscreen directly (if needed).

**Comment by lethosor** [comment]
This completely breaks Stonesense, because Stonesense is garbage. Not your fault - apparently there's an X11 Screen struct, and Stonesense has `using namespace DFHack` in several places before including X11 headers.

**Comment by suokko** [comment]
The stonesense change that I originally forgot to commit and push and then link to here: https://github.com/DFHack/stonesense/pull/59

**Comment by lethosor** [comment]
Oops. Well, hopefully it'll be more resilient to future changes that conflict with X11/Allegro/whatever

**Pull request merged**